src/ser/mod.rs

```diff
@@ -378,6 +383,15 @@ pub struct Serializer<W: fmt::Write> {
     implicit_some_depth: usize,
 }
 
+fn indent<W: fmt::Write>(mut output: W, config: &PrettyConfig, pretty: &Pretty) -> fmt::Result {
```

Comment:
Could this take a mutable reference to output instead, since we're anyways always passing one?